apps/examples/testcase/le_tc/compression/tc_compress_read.c

```diff
@@ -0,0 +1,88 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <debug.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <tinyara/config.h>
+#include <tinyara/lzma/LzmaLib.h>
+#include <tinyara/fs/fs.h>
+#include <tinyara/binfmt/compression/compress_read.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <stdint.h>
+#include <unistd.h>
+#include <string.h>
+#include <tc_common.h>
+
+/****************************************************************************
+ * Private Declarations
+ ****************************************************************************/
+
+struct s_header *compression_header;
+struct s_buffer buffers;
+
+/****************************************************************************
+ * Public Function
+ ****************************************************************************/
+int tc_compress_read_main(void)
+{
+	int i;
+	int filefd;
+	int ret;
+	off_t filelen;
+	unsigned int writesize;
+	unsigned int size;
+
+	filefd = open("/mnt/myfile_comp", O_RDONLY);
+
+	if (filefd == (-1)) {
+		int errval = get_errno();
+		berr("Failed to open file ERROR = %d\n", errval);
+		return -errval;
+	}
+
+	ret = compress_init(filefd, 0, &filelen);
+	if (ret != OK) {
+		berr("Failed to read header for compressed binary : %d\n", ret);
+		return ret;
+	}
+
+	for (i = 0; i < compression_header->sections - 1; i++) {
+		size = compress_read_block(filefd, 0, buffers.read_buffer, i);
+		if (size < 0) {
+			bmdbg("Read for compressed block %d failed\n", i);
```

Comment:
Why don't you add TC_ASSERT_XXX macro for checking this tc?

Reply:
Changes made.